File: src/002.String/14.LongestCommonPrefix/14.最长公共前缀.js
/*
 * @lc app=leetcode.cn id=14 lang=javascript
 *
 * [14] 最长公共前缀
 */

// @lc code=start
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function (strs) {
	if (!strs || !strs.length) return '';

	for (let i = 0; i < strs[0].length; i++) {
		for (let j = 0; j < strs.length; j++) {
			if (i == strs[j].length || strs[0][i] != strs[j][i]) {
				return strs[0].substring(0, i);
			}
		}
	}

	return strs[0];
};
// @lc code=end

File: src/002.String/14.LongestCommonPrefix/14.最长公共前缀.go
/*
 * @lc app=leetcode.cn id=14 lang=golang
 *
 * [14] 最长公共前缀
 */

// @lc code=start
func longestCommonPrefix(strs []string) string {
	result := strs[0]

	for _, v := range strs {
		for strings.Index(v, result) != 0 {
			if len(result) == 0 {
				return ""
			}
			result = result[:len(result)-1]
		}
	}

	return result
}

// @lc code=end

